pkg/gjson/row.go

```diff
@@ -217,15 +234,48 @@ func constructColumnTree(result gjson.Result, originalQuery string) *columnTree
 	return ct
 }
 
+func applyOptions(tree *columnTree, options ...ColumnTreeOptions) {
+	for _, option := range options {
+		option(tree)
+	}
+}
+
 func (ct *columnTree) CreateColumns() [][]string {
 	// Get the number of queries. Each query represents a column.
 	numberOfQueries := getNumberOfQueries(ct.originalQuery)
 	columns := make([][]string, 0, numberOfQueries)
+	var strictColIndices []int
+	if len(ct.strictColumns) > 0 {
+		for i, r := range ct.result {
+			if slices.Contains(ct.strictColumns, r.query) {
+				strictColIndices = append(strictColIndices, i)
+			}
+		}
+	}
+	deletionMap := make(map[int]bool)
```

Comment:
I think ` := set.IntSet{}` will be more explicit. With bool it looks like we care if key is in the map and what value it has.

Reply:
A set was what I was looking for, I thought Go doesn't have them. Will change to this 👍